spark/src/main/scala/org/apache/spark/sql/comet/CometSparkToColumnarExec.scala

```diff
@@ -85,14 +103,30 @@ case class CometSparkToColumnarExec(child: SparkPlan)
     } else {
       child
         .execute()
-        .mapPartitionsInternal { iter =>
+        .mapPartitionsInternal { sparkBatches =>
           val context = TaskContext.get()
-          CometArrowConverters.rowToArrowBatchIter(
-            iter,
-            schema,
-            maxRecordsPerBatch,
-            timeZoneId,
-            context)
+          val arrowBatches =
+            CometArrowConverters.rowToArrowBatchIter(
+              sparkBatches,
+              schema,
+              maxRecordsPerBatch,
+              timeZoneId,
+              context)
+
+          new Iterator[ColumnarBatch] {
+
+            override def hasNext: Boolean = {
+              arrowBatches.hasNext
+            }
+
+            override def next(): ColumnarBatch = {
+              // The conversion happens in next(), so redefine the call to measure time spent.
+              val startNs = System.nanoTime()
+              val batch = arrowBatches.next()
+              conversionTime += System.nanoTime() - startNs
+              batch
+            }
+          }
```

Comment:
I see that we crete this iterator for both child and columnar inputs. We could avoid duplicating the code and call a method to create this iterator and then call this from both branches.

```scala
  private def createTimingIter(iter: Iterator[ColumnarBatch], conversionTime: SQLMetric): Iterator[ColumnarBatch] = {
    new Iterator[ColumnarBatch] {

      override def hasNext: Boolean = {
        iter.hasNext
      }

      override def next(): ColumnarBatch = {
        val startNs = System.nanoTime()
        val batch = iter.next()
        conversionTime += System.nanoTime() - startNs
        batch
      }
    }
  }
```